### PR TITLE
release(required): Fix logger storage initialization errors

### PR DESCRIPTION
**Pull request by yuhengshs**
#### Description of changes

Fix the logger `error` from PR https://github.com/aws-amplify/amplify-js/pull/13938

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Unit Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

#### Checklist for repo maintainers

- [ ] Verify E2E tests for existing workflows are working as expected or add E2E tests for newly added workflows
- [ ] New source file paths included in this PR have been added to CODEOWNERS, if appropriate

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/storage/utils.ts [hunk 1]
```diff
@@ -44,7 +44,9 @@ export const getSessionStorageWithFallback = (): Storage => {
 		throw new Error('sessionStorage is not defined');
 	} catch (e) {
 		// Handle any errors related to sessionStorage access
-		logger.error('SessionStorage access failed:', e);
+		logger.info(
+			'sessionStorage access failed. InMemoryStorage is used as a fallback.',
```
Comment (sobolk) [inline packages/core/src/storage/utils.ts:48]:
```suggestion
			'SessionStorage not found. InMemoryStorage is used as a fallback.',
```

Comment (sobolk) [reply packages/core/src/storage/utils.ts:48]: we should avoid, error / failure language, this is valid on node.

File: packages/core/src/storage/utils.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export const getLocalStorageWithFallback = (): Storage => {
 		}
 	} catch (e) {
 		// Handle any errors related to localStorage access
-		logger.error('LocalStorage access failed:', e);
+		logger.info('localStorage access failed. InMemoryStorage is used as a fallback.');
```
Comment (sobolk) [inline packages/core/src/storage/utils.ts:23]:
```suggestion
		logger.info('LocalStorage not found. InMemoryStorage is used as a fallback.');
```

File: packages/core/polyfills/URL/index.js [hunk 3]
```diff
@@ -0,0 +1 @@
+(()=>{var e,r={515:e=>{"use strict";e.exports=require("react-native-url-polyfill/auto")}},o={},t={};"test"!==(null===(e=null===process||void 0===process?void 0:process.env)||void 0===e?void 0:e.NODE_ENV)&&function e(t){var s=o[t];if(void 0!==s)return s.exports;var i=o[t]={exports:{}};return r[t](i,i.exports,e),i.exports}(515);var s=exports;for(var i in t)s[i]=t[i];t.__esModule&&Object.defineProperty(s,"__esModule",{value:!0})})();
```
Comment (dbanksdesign) [inline packages/core/polyfills/URL/index.js:1]: Is this supposed to be part of the PR?

Comment (yuhengshs) [reply packages/core/polyfills/URL/index.js:1]: This is an extra file, removing it on next commit

**Review by sobolk — APPROVED** [review]

**Review by cwomack — APPROVED** [review]

**Review by ashika112 — APPROVED** [review]

**Review by jjarvisp — APPROVED** [review]

**Review by pranavosu — APPROVED** [review]

**Pull request merged**